File: SpeedTestGameClient/core/src/main/java/si/um/feri/speedii/classes/ExtremeEvent.java
package si.um.feri.speedii.classes;

import java.time.LocalDateTime;

public class ExtremeEvent {
    public Events type;
    public Location location;
    public LocalDateTime time;
    public User user;

    public ExtremeEvent(Events type, Location location, LocalDateTime time, User user) {
        this.type = type;
        this.location = location;
        this.time = time;
        this.user = user;
    }

    public String toString() {
        return "Event: " + type + " at " + location + " at " + time + " by " + user;
    }
}

File: MobileApp/app/speedtest/src/main/java/sosteric/pora/speedtest/ExtremeEvent.kt
package sosteric.pora.speedtest

import Location
import User
import java.time.LocalDateTime

class ExtremeEvent(
    val type: Events,
    val location: Location,
    val time: LocalDateTime,
    val user: User?
) {

    override fun toString(): String {
        return "$type, $location, $time, $user"
    }

    class ExtremeEventAlt(
        val type: String,
        val location: Location,
        val time: LocalDateTime,
        val user: String?
    )

    fun toAlt(): ExtremeEventAlt {
        val currentUser = user
        val userId = if(user != null) { currentUser!!.id.toString() } else { null }
        return ExtremeEventAlt(type.toString(), location, time, userId)
    }
}